### DISCUSSION TITLE
Flag for preventing outside click behaviour

### DISCUSSION THREAD

**Discussion by agcty** [tailwindlabs/headlessui#1157] [Ideas]
The Dialog component is great but we're running into a lot of headaches when it comes to outside click behavior as described in #621. Especially working with toasts is a real pain at the moment and a lot of people have also seem to have similar issues.

**Proposal**:

Add a flag that lets you disable closing the dialog on outside click

#768 could also work but an explicit flag might make it more developer-friendly

**Comment by soulsam480** [comment]
This will be superr helpful
